packages/react-instantsearch-core/src/lib/__use.ts

```diff
@@ -0,0 +1,32 @@
+// Taken from https://github.com/apollographql/apollo-client/blob/ba3e7d9fa7d46e4c636148bbf01552833db0ceda/src/react/hooks/internal/__use.ts
+
+/* eslint-disable default-case */
+/* eslint-disable @typescript-eslint/naming-convention */
+import * as React from 'react';
+
+import { wrapPromiseWithState } from './wrapPromiseWithState';
+
+type Use = <T>(promise: Promise<T>) => T;
+// Prevent webpack from complaining about our feature detection of the
+// use property of the React namespace, which is expected not
+// to exist when using current stable versions, and that's fine.
+const useKey = 'use' as keyof typeof React;
+const realHook = React[useKey] as Use | undefined;
+
+// This is named with two underscores to allow this hook to evade typical rules of
+// hooks (i.e. it can be used conditionally)
+export const __use =
```

Comment:
we can't get rid of the fallback without a breaking change, as older react versions (that don't use server side rendering for example) don't have use

Reply:
It won't get called unless it's wrapped by `InstantSearchRSCContext` though

Reply:
in that case we probably should directly use React.use (not import as that probably errors in some cases with old react as it's outside of the conditional?)

Reply:
Can't do `React.use` either as it's not in the types, but I kept the `React['use' as keyof typeof React]` and it works well